src/app/archive/archive_lib/processor.ml

```diff
@@ -1355,34 +1355,57 @@ module Block = struct
     else return ()
 end
 
-let add_block_aux ~add_block ~hash ~delete_older_than
+let retry ~f ~logger ~error_str retries =
+  let rec go retry_count =
+    match%bind f () with
+    | Error e ->
+        if retry_count <= 0 then return (Error e)
```

Comment:
maybe accumulate errors?

Reply:
In addition to logging it as warnings (the else case)? The error in this line is logged at the call site

Reply:
probably good enough